examples/DynamoDB_Table_With_KinesisStreamSpecification.py

```diff
@@ -0,0 +1,106 @@
+# Converted from DynamoDB_Table.template located at:
+# http://aws.amazon.com/cloudformation/aws-cloudformation-templates/
+
+from troposphere import Output, Parameter, Ref, Template
+from troposphere.dynamodb import (
+    AttributeDefinition,
+    KeySchema,
+    KinesisStreamSpecification,
+    ProvisionedThroughput,
+    Table,
+)
+
+t = Template()
+
+t.set_description(
+    "Create a DynamoDB Table with a Kinesis Stream Specification."
+)
+
+hashkeyname = t.add_parameter(
+    Parameter(
+        "HashKeyElementName",
+        Description="HashType PrimaryKey Name",
+        Type="String",
+        AllowedPattern="[a-zA-Z0-9]*",
+        MinLength="1",
+        MaxLength="2048",
+        ConstraintDescription="must contain only alphanumberic characters",
+    )
+)
+
+hashkeytype = t.add_parameter(
+    Parameter(
+        "HashKeyElementType",
+        Description="HashType PrimaryKey Type",
+        Type="String",
+        Default="S",
+        AllowedPattern="[S|N]",
+        MinLength="1",
+        MaxLength="1",
+        ConstraintDescription="must be either S or N",
+    )
+)
+
+readunits = t.add_parameter(
+    Parameter(
+        "ReadCapacityUnits",
+        Description="Provisioned read throughput",
+        Type="Number",
+        Default="10",
+        MinValue="5",
+        MaxValue="10000",
+        ConstraintDescription="should be between 5 and 10000",
+    )
+)
+
+writeunits = t.add_parameter(
+    Parameter(
+        "WriteCapacityUnits",
+        Description="Provisioned write throughput",
+        Type="Number",
+        Default="5",
+        MinValue="5",
+        MaxValue="10000",
+        ConstraintDescription="should be between 5 and 10000",
+    )
+)
+
+streamarn = t.add_parameter(
+    Parameter(
+        "StreamArn",
+        AllowedPattern="arn:aws:\\S+:\\S+:\\d+:\\w+\\/\\S+",
```

Comment:
Would appreciate any feedback on this line. I wasn't sure if this should have been more simplified or even just `.*`

Reply:
The above pattern should work. @PatMyron do you know of a ARN regex that would be better to use?

Reply:
As mentioned in https://github.com/aws-cloudformation/cloudformation-cli/pull/675, `arn:aws:` should instead be something like `arn:aws[-a-z]*:` to support [other partitions](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax)

Reply:
Awesome. Updated. Thank you for the feedback.

Reply:
I believe @PatMyron was implying the regex should be:
"arn:aws[-a-z]*:\\S+:\\S+:\\d+:\\w+\\/\\S+"

Reply:
@markpeek @PatMyron Oh. I looked at the links provided and didn't see any examples using the word or digit metacharacter and assuming he was suggesting I simplify it. But I see that the MR referenced was for extend partition support. I'd be happy to change it back to `arn:aws[-a-z]*:\S+:\S+:\d+:\w+\/\S+"`. Is that a good pattern?

Reply:
I might put `.*` at the end for the resource name (after the final ":") but looks fine after that change.

Reply:
:+1: -- Just updated to: `arn:aws[-a-z]*:\\S+:\\S+:\\d+:.*`. Thanks again